src/nodes.coffee

```diff
@@ -1187,6 +1197,28 @@ exports.Obj = class Obj extends Base
       prop = prop.unwrapAll()
       prop.eachName iterator if prop.eachName?
 
+  # Object spread properties. https://github.com/tc39/proposal-object-rest-spread/blob/master/Spread.md
+  # obj2 = {a:1, obj..., c:3, d:4} => obj2 = Object.assign({}, {a:1}, obj1, {c:3, d:4})
+  compileSpread: (o) ->
+    props = @properties
+    # Store object spreads.
+    splatSlice = []
+    propSlices = []
+    slices = [new Obj [], false]
+    addSlice = () -> 
+      slices = [slices..., new Obj [propSlices...], false] if propSlices.length      
```

Comment:
There's no need to slice `propSlices` since it is reassigned (e.g. `[propSlices...]` could just be `propSlices`).

Also, `generated = no` by default, so the second argument to `new Obj` is unnecessary.

Also, this might be a bit more readable (and performant :man_shrugging:) if it used `.push` instead of `...`, same with below. Personally I like the 'immutability' of using spreads like this, but it's the common style in the codebase to use `.push args...` instead.